shell/common/snapshot_controller_impeller.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/snapshot_controller_impeller.h"
+#include <algorithm>
+
+#include "flutter/flow/surface.h"
+#include "flutter/fml/trace_event.h"
+#include "flutter/impeller/display_list/display_list_dispatcher.h"
+#include "flutter/impeller/display_list/display_list_image_impeller.h"
+#include "flutter/impeller/geometry/size.h"
+#include "flutter/shell/common/snapshot_controller.h"
+
+namespace flutter {
+
+sk_sp<DlImage> SnapshotControllerImpeller::MakeRasterSnapshot(
```

Comment:
This method seems like it should be guarded by the GPU sync switch...

Reply:
Done.